### PR TITLE
Happy thoughts API - Ingela

### PR DESCRIPTION
**Pull request by IngelaL**
It was a fun project and really interesting to do the backend part and to connect it to the frontend. I think the material and lectures gave us the tools needed to do it. 

Heroku: https://happy-thoughts-ingela.herokuapp.com/
Netlify:  https://ingelas-happy-thoughts.netlify.app/

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,13 +1,32 @@
...
+Endpoints:
+
+https://happy-thoughts-ingela.herokuapp.com/thoughts 
+Here you  will get a list of the 20 most recent thoughts. If you do a POST to this endpoint you will need a JSON body with the thought "message", like this: `{ "message": "I am feeling great!" }`. If the input is valid, which means 5-140 characters, the thought should be saved, and the response should include the saved thought object, including its `_id`.
+
+https://happy-thoughts-ingela.herokuapp.com/thoughts/:thoughtId/like
+This endpoint doesn't require a JSON body. Given a valid thought id in the URL, the API should find that thought, and update its hearts property to add one heart.
+
+Here you can find the frontend part of this project and try it out: https://ingelas-happy-thoughts.netlify.app/ (and here is my git hub repo for that project: https://github.com/IngelaL/project-happy-thoughts)
+
```
Comment (MindstormingAB) [inline README.md:32]: Nice description/API documentation

File: server.js [hunk 2]
```diff
@@ -20,7 +37,42 @@ app.use(bodyParser.json())
...
+app.post('/thoughts', async (req, res) => {
+  //Retrieve the information sent by the client to our API endpoint
+  const {message} = req.body
+
+  //Use our mpngoose model to create the database entry
+  const thoughts = new Thought({message})
+
+  try {
+    //Success - the thought is saved to the databse
+    const savedThought = await thoughts.save()
```
Comment (MindstormingAB) [inline server.js:57]: Be mindful of the naming of your constant. You are creating 1 thought, `const thought` would have been more pertinent

File: server.js [hunk 3]
```diff
@@ -20,7 +37,42 @@ app.use(bodyParser.json())
...
+    res.status(400).json({message: 'Could not save thought', error: err.errors})
+    }
+})
+
+app.post('/thoughts/:thoughtId/like', async (req, res) => {
+  const {thoughtId} = req.params
+
+  try {
+    // Success - the thought will be updated with a heart like
+    const thoughtLiked = await Thought.updateOne({ _id: thoughtId }, { $inc: { hearts: 1 } });
```
Comment (MindstormingAB) [inline server.js:70]: Great!
For your information, the method updateOne() is doing the job but the response in Postman is a bit cryptic. If you use the method findOneAndUpdate(), you can get the updated document as response.
```suggestion
    const thoughtLiked = await Thought.findOneAndUpdate({ _id: thoughtId }, { $inc: { hearts: 1 }, { new: true } });
```

**Review by MindstormingAB — APPROVED** [review]
Well done Ingela!
I visited your API on Heroku and I tested your endpoints in the frontend and with Postman.
They behave as expected for both correct and bad requests.
Your code is clean, your comments helpful, you easily reached the blue level. 🥳

**Review by KarolCislowski — APPROVED** [review]
Very nice and clean app 👍🏻

**Pull request closed without merging**